src/compiler/utilities.ts

```diff
@@ -215,8 +215,37 @@ namespace ts {
 
     // Gets the nearest enclosing block scope container that has the provided node
     // as a descendant, that is not the provided node.
-    export function getEnclosingBlockScopeContainer(node: Node): Node {
-        let current = node.parent;
+    export function getEnclosingBlockScopeContainer(node: Node): Node;
+
+    /**
+      * Gets the nearest enclosing block scope container that has the provided node
+      * as a descendant, that is not the provided node.
+      * @param node The starting node
+      * @param getAncestorOrSelf A callback used to get the ancestor of the starting node, used 
+      * only when traversing the ancestors of the current node in the emitter.
+      * @param offset The offset in the node stack used to get the ancestor of the current node

```

Comment:
This phrasing is a bit awkward to me, personally. Would it be possible to phrase it as
"The depth [or height?] in the node stack to ascend from 'node'."
or
"The number of parent-pointers to traverse"?

For the latter, it might be worth noting additionally that we don't actually use the parent pointers to do the traversal, as appears to be the point of this PR :)